src/v1/imports/carts/grassdoor.ts

```diff
@@ -1,86 +1,62 @@
-  const dataLayer = window.dataLayer || [];
-
-  function onDataLayerChange() {
-    const data = window.dataLayer.slice(-1)[0]; // Gets the newest array member of dataLayer
-
-    if (data.event === "Product Added") {
-      const products = data;
-      const { sku, name, price, quantity, category } = products;
-
+  grassDoorTracker({
+    addToCartEvent(addToCartData) {
       window.tracker(
         "trackAddToCart",
-        sku.toString(),
-        (name || "N/A").toString(),
-        (category || "N/A").toString(),
-        parseFloat(price || 0),
-        parseInt(quantity || 1),
-        "USD"
+        addToCartData.sku,
+        addToCartData.name,
+        addToCartData.category,
+        addToCartData.unitPrice,
+        addToCartData.quantity,
+        addToCartData.currency
       );
-    }
-
-    if (data.event === "Product Removed") {
-      const products = data;
-      const { sku, name, price, quantity, category } = products;
+    },
 
+    removeFromCartEvent(cartData) {
       window.tracker(
         "trackRemoveFromCart",
-        sku.toString(),
-        (name || "N/A").toString(),
-        (category || "N/A").toString(),
-        parseFloat(price || 0),
-        parseInt(quantity || 1),
-        "USD"
+        cartData.sku,
+        cartData.name,
+        cartData.category,
+        cartData.unitPrice,
+        cartData.quantity,
+        cartData.currency
       );
-    }
-
-    if (data.event === "Order Made") {
-      const transaction_id = data.order_id;
-      const transaction_total = data.revenue;
-      const transaction_tax = data.tax;
-      const transaction_shipping = data.shipping;
-      const products = data.products;
+    },
 
+    transactionEvent(transactionData) {
       window.tracker(
         "addTrans",
-        transaction_id.toString(),
+        transactionData.id,
         retailId ?? appId,
-        parseFloat(transaction_total || 0),
-        parseFloat(transaction_tax || 0),
-        parseFloat(transaction_shipping || 0),
-        "N/A",
-        "N/A",
-        "N/A",
-        "USD"
+        transactionData.total,
+        transactionData.tax,
+        transactionData.shipping,
+        transactionData.city,
+        transactionData.state,
+        transactionData.country,
+        transactionData.currency
       );
 
-      products.forEach((items) => {
-        const { product_id, name, price, quantity, category } = items;
-
+      transactionData.items.forEach((item) => {
         window.tracker(
           "addItem",
-          transaction_id.toString(),
-          product_id.toString(),
-          (name || "N/A").toString(),
-          (category || "N/A").toString(),
-          parseFloat(price || 0),
-          parseInt(quantity || 1),
-          "USD"
+          transactionData.id,
+          item.productId,
+          item.sku,
+          item.name,
+          item.category,
+          item.unitPrice,
+          item.quantity,
+          item.currency
```

Comment:
once again, review the snowplow schema for `addItem` since this is incorrect.

https://docs.google.com/document/d/1VAAjSJw618fq1BY7t7hJTcncItIDi2TF/edit